reascripts/ReaSpeech/source/SettingsPlugin.lua

```diff
@@ -0,0 +1,14 @@
+--[[
+
+  SettingsPlugin.lua - plugin to manage basic ReaSpeech settings
+
+]]--
+
+SettingsPlugin = Plugin {}
+
+function SettingsPlugin:init()
+  assert(self.app, 'SettingsPlugin: plugin host app is required')
+  Logging.init(self, 'SettingsPlugin')
+  self._controls = SettingsControls.new(self)
+  self._actions = SettingsActions.new(self)
+end
```

Comment:
> Assuming Settings grows to handle persistence at some point, how does the state management tie into this? Do we need a concept of a model in this plugin structure?

i hadn't personally settled on how it would look specifically, but off the top of my head i can see some concept of a persistent store that's available for the main app or for plugins to use on an ad-hoc basis. there's a couple of connection points off the top of my head that could be useful, like:
 - a basic get/set interface (`<plugin instance>.app.persistence:get(<key>)`)
 - extra args to stateful widgets

i can also imagine a world where "settings" becomes part of the core (or, depending on what we allow wrt interaction between plugins some kind of privileged plugin). other plugins could be queried for any settings options that they provide. content to let that concept bake for awhile longer.